### PR TITLE
Remove old documentation from docs directory

### PR DESCRIPTION
**Pull request by russcam**
This commit removes the old generated documentation from the docs directory, and adds in new documentation to explain serialization.

### CODE REVIEW

File: docs/client-concepts/high-level/serialization/custom-serialization.asciidoc [hunk 1]
```diff
@@ -13,36 +13,41 @@ please modify the original csharp file found at the link and submit the PR with
...
-NEST has always isolated Json.NET as best as it could but this meant that we had to mandate some things.
-For instance NEST heavily relied on the fact that the `ContractConverter` was an instance of `ElasticContractConverter`
-
-If you wanted to deserialize your `_source` or `_fields` using your own `ContractConverter` you were out of luck.
+Why would we do this, you may ask? Well, NEST has always isolated its dependency on Json.NET as best as it could,
+but this meant that we had to mandate how certain things were in the client. For instance,
+NEST heavily relied on the fact that the `IContractResolver` used by the configured serializer was
+an instance of `ElasticContractResolver`, so if you wanted to deserialize your `_source` or `_fields`
+using your own resolver, you were out of luck. In addition, continued improvements to NEST's serialization pipeline
+was stymied by this dependency and as client authors, we wanted to unhinder ourselves from this in order to explore the myriad
```
Comment (codebrain) [inline docs/client-concepts/high-level/serialization/custom-serialization.asciidoc]: :+1: stymied!

File: docs/client-concepts/high-level/serialization/custom-serialization.asciidoc [hunk 2]
```diff
@@ -13,36 +13,41 @@ please modify the original csharp file found at the link and submit the PR with
...
`JsonSerializerSettings` and `JsonConverter` types for example, things have changed a bit. The following section explains
+how to continue working with these with NEST 6.x.
 
 [float]
-=== Injecting a new serializer
+==== Injecting a new serializer
 
 Starting with NEST 6.x you can inject a serializer that is isolated to only be called
-for the (de)serialization of `_source` `_fields` and where ever a user provided value is expected
+for the (de)serialization of `_source`, `_fields`, or whereever a user provided value is expected
```
Comment (codebrain) [inline docs/client-concepts/high-level/serialization/custom-serialization.asciidoc]: wherever! ;)

File: docs/client-concepts/high-level/serialization/custom-serialization.asciidoc [hunk 3]
```diff
@@ -124,21 +137,18 @@ the contract resolver from `Json.NET`.
 ----
 var pool = new SingleNodeConnectionPool(new Uri("http://localhost:9200"));
 var connectionSettings =
-    new ConnectionSettings(pool, sourceSerializer: (b, s) => new JsonNetSerializer(
-        b, s,
+    new ConnectionSettings(pool, sourceSerializer: (builtin, settings) => new JsonNetSerializer(
+        builtin, settings,
```
Comment (codebrain) [inline docs/client-concepts/high-level/serialization/custom-serialization.asciidoc]: :+1: for removing the BS

**Comment by Mpdreamz** [comment]
ported to `6.x`

**Pull request merged**